src/pool.c

```diff
@@ -1953,7 +1953,15 @@ rpc_on_wallet_transferred(const char* data, rpc_callback_t *callback)
             continue;
         }
         uint64_t current_amount = *(uint64_t*)val.mv_data;
-        current_amount -= payment->amount;
+
+        if (current_amount >= payment->amount) {
+            current_amount -= payment->amount;
+        } else {
+            log_error("Payment was more than balance: %"PRIu64" > %"PRIu64,
+                      payment->amount, current_amount);
+            current_amount = 0;
```

Comment:
If the payout amount is greater than the current balance, which should never be the case but I still like the sanity check, we probably should not touch the existing balance or make a payment or anything else for this payment (other than a error log of course). Maybe just jump to cleanup after the log.

Reply:
Actually, I'm wrong, we've already made the payout by this point. Yes setting to zero makes sense (with the log).

Reply:
At this point a payment was successfully made, so I think it should be set to 0.

Reply:
Thank you again.